orttraining/orttraining/core/framework/mpi_setup.h

```diff
@@ -19,7 +21,7 @@ struct MPIContext {
   int local_size;
 };
 
-#if defined(USE_NCCL) || defined(USE_HOROVOD)
+#if defined(USE_MPI)
```

Comment:
The send and recv kernels are guarded by USE_HOROVOD, I think we should change that to use this directive especially for the cpu kernels.